src/framegrab/grabber.py

```diff
@@ -1035,6 +1045,50 @@ def _apply_camera_specific_options(self, options: dict) -> None:
             pass
 
 
+class RaspberryPiCSI2FrameGrabber(FrameGrabber):
+    "For CSI2 cameras connected to Raspberry Pis through their dedicated camera port"
+
+    def __init__(self, config: dict):
+        self.config = config
+
+        # This will also detect USB cameras, but according to the documentation CSI2
+        # cameras attached to the dedicated camera port will always come before USB
+        # cameras in the resulting list of camera dictionaries
+        # https://datasheets.raspberrypi.com/camera/picamera2-manual.pdf#page=66.21
+        cameras = Picamera2.global_camera_info()
+
+        if not cameras:
+            raise ValueError("No CSI2 cameras were found. Is your camera connected?")
+
+        # Since global_camera_info() also finds USB cameras, we will only use the first
+        # entry. USB cameras must be found through their specific FrameGrabber. Note
+        # that only a single CSI2 camera is supported at this time.
+        picam2 = Picamera2()
```

Comment:
This seems a little weird to me, but I guess it's just how picamera2 works? Is `Picamera2()` guaranteed to return the same camera as `Picamera2.global_camera_info()[0]`? I think you mentioned that you had tested this with a USB camera plugged in and it worked. In that case, it should be fine.

Reply:
I did test it with a USB camera plugged in. Picamera doesn't fully support USB cameras, so if a CSI2 camera is available it always uses that one. It's documented explicitly for `global_camera_info`, and `Picamera2` takes an optional camera num argument, which defaults to 0. So same deal, it'll always get the first one (which will be the CSI2 camera if one is connected). I took a look at the [source code](https://github.com/raspberrypi/picamera2/blob/07b3f23ebeaa58a45f04ce1dd62aa45db03e1d6a/picamera2/picamera2.py#L216) to verify.